orca-whirlpool/src/modules/4_map_deposits.rs

```diff
@@ -0,0 +1,67 @@
+use crate::{
+    key_store::StoreKey,
+    pb::messari::orca_whirlpool::v1::{event::Type, Deposit, Deposits, Events, Pool},
+};
+use substreams::{
+    log, skip_empty_output,
+    store::{StoreGet, StoreGetProto},
+};
+
+#[substreams::handlers::map]
+pub fn map_deposits(
+    raw_events: Events,
+    pools_store: StoreGetProto<Pool>,
+) -> Result<Deposits, substreams::errors::Error> {
+    skip_empty_output();
+
+    let zero = "0".to_string();
+
+    let data: Vec<Deposit> = raw_events
+        .data
+        .into_iter()
+        .filter_map(|event| {
+            if let Some(Type::IncreaseLiquidity(increase_liquidity_event)) = event.r#type {
+                let id = format!("{}-{}", event.txn_id, event.slot);
+
+                let accounts = increase_liquidity_event.accounts?;
+                let instruction = increase_liquidity_event.instruction?;
+
+                let pool = pools_store.get_last(StoreKey::Pool.get_unique_key(&accounts.whirlpool));
+
+                if pool.is_none() {
+                    log::info!("Pool not found: {:?}", accounts.whirlpool);
+                    return None;
+                }
+                let pool = pool.unwrap();
```

Comment:
Unwrap can simplify the code a lot, but it also makes the program prone to crashing or having bugs that are difficult to identify. There's a middle ground for doing this vs doing all of the handling. As long as you are aware of the tradeoff.